### PR TITLE
termsnap: init at 0.3.0

### PR DESCRIPTION
**Pull request by yash-garg**
## Description of changes

Added [termsnap](https://github.com/tomcur/termsnap) package which is used to create SVGs from terminal output.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by pbsds** [comment]
This change is missing an addition to `all-packages.nix`, which is now discouraged. Please migrate to by-name

**Comment by Yash-Garg** [comment]
> This change is missing an addition to `all-packages.nix`, which is now discouraged. Please migrate to by-name

done!

File: pkgs/by-name/te/termsnap/package.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "termsnap";
+  version = "0.3.0";
+
+  src = fetchFromGitHub {
+    owner = "tomcur";
+    repo = pname;
+    rev = "${pname}-v${version}";
```
Comment (afh) [inline pkgs/by-name/te/termsnap/package.nix:14]: Superfluous use of `pname` is frowned upon (for details see #277994):
```suggestion
    repo = "termsnap";
    rev = "termsnap-v${version}";
```

**Review by afh — CHANGES_REQUESTED** [review]
Thanks for this, @Yash-Garg, please find below a suggestion to improve the package 🙂

**Comment by afh** [comment]
Thanks for addressing the review feedback so promptly, @Yash-Garg, much appreciated.

A few more things:

1. Please add yourself to the [`maintainers/maintainer-list.nix`](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix) in a separate single commit. For an example see ac189de1e50472903ab82a43880953130b3d500b.
For more details please refer to [Commit conventions in CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions) and the [Nixpkgs maintainers README](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md)
2. Squash all existing commits into a single one with the message `termsnap: init at 0.3.0`
3. Make sure that the commit adding you to the maintainers list comes before the commit that adds the new termsnap package

Hopefully I've been able to explain well what is needed to follow nixpkgs' best practices, if not please ask me to clarify 🙂

**Comment by Yash-Garg** [comment]
> 1. ac189de

Thanks, made the required changes.

**Comment by afh** [comment]
Result of `nixpkgs-review pr 327997` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>termsnap</li>
  </ul>
</details>

**Review by afh — APPROVED** [review]
Thanks for addressing the suggestions so swiftly, @Yash-Garg, much appreciated 👍 

If you wanted to go above and beyond, feel free to change the first commit message to `maintainers: add yash-garg` (notice the added `s` to maintainers). Unfortunately the example I gave did not follow what the majority of other commits to the maintainers-list.nix file do 😕  My bad.

And leaving the commit message as it currently is, is also fine 🙂

**Comment by pbsds** [comment]
welcome to nixpkgs!

**Comment by afh** [comment]
Nice thorough review and checklist, @pbsds, is this part of your personal review best practices or newer versions of nixpgs-review?

**Comment by pbsds** [comment]
I use [this list](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#new-packages) at times to make sure i don't start glossing over important things.

**Comment by afh** [comment]
That's very helpful, thanks, @pbsds 🙂 👍

**Pull request merged**